exoscale/instances.go

```diff
@@ -93,10 +94,30 @@ func (i *instances) NodeAddressesByProviderID(ctx context.Context, providerID st
 		return nil, err
 	}
 
-	return []v1.NodeAddress{{
+	addresses := []v1.NodeAddress{{
 		Type:    v1.NodeExternalIP,
 		Address: instance.PublicIPAddress.String(),
-	}}, nil
+	}}
+
+	if i.p.client != nil && instance.PrivateNetworkIDs != nil && len(*instance.PrivateNetworkIDs) > 0 {
+		if node, _ := i.p.kclient.CoreV1().Nodes().Get(context.TODO(), *instance.Name, metav1.GetOptions{}); node != nil {
```

Comment:
Is there any reason for not using `ctx`?

```suggestion
		if node, _ := i.p.kclient.CoreV1().Nodes().Get(ctx, *instance.Name, metav1.GetOptions{}); node != nil {
```

Reply:
Oh, sorry, i miss that
I guess this is copy/past issue )